### PR TITLE
Register annexes

### PR DESCRIPTION
**Pull request by meaboutsoftware**
This PR contains changes related to Annexes:

- Add private ctor that is required by EF
- Change return type on Binding contract while adding annex to void
- Add ownership of Annexes to the Binding contract
- Database migration (Binding Contract owns many Annexes)

### CODE REVIEW

File: Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/BindingContract.cs [hunk 1]
```diff
@@ -42,12 +44,12 @@ public sealed class BindingContract : Entity
         DateTimeOffset bindingFrom,
         DateTimeOffset expiringAt) => new(id, customerId, duration, bindingFrom, expiringAt);
 
-    public Annex AddAnnex(DateTimeOffset validFrom, DateTimeOffset now)
+    public void AddAnnex(DateTimeOffset validFrom, DateTimeOffset now)
```
Comment (kamilbaczek) [inline Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/BindingContract.cs:47]: What you think to rename this method to "Attach"?

I've checked some business conversations in the Internet and I think is more appropriate.

```suggestion
    public void AttachAnnex(DateTimeOffset validFrom, DateTimeOffset now)
```

File: Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/BindingContract.cs [hunk 2]
```diff
@@ -42,12 +44,12 @@ public sealed class BindingContract : Entity
...
     {
         BusinessRuleValidator.Validate(
             new AnnexCanOnlyBeAddedOnlyBeAddedToActiveBindingContractRule(TerminatedAt, ExpiringAt, now));
 
-        return Annex.Add(Id, validFrom);
+        RegisteredAnnexes.Add(Annex.Add(Id, validFrom));
```
Comment (kamilbaczek) [inline Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/BindingContract.cs:52]: We could change RegisteredAnnexes to be similar name to method that is defining it.

var annex = Annex.Create(Id, validFrom);
AttachedAnnexes.Add(annex);

What you think?

File: Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Infrastructure/Database/Configurations/BindingContractEntityConfiguration.cs [hunk 3]
```diff
@@ -27,5 +27,23 @@ public void Configure(EntityTypeBuilder<BindingContract> builder)
         builder.Property(contract => contract.TerminatedAt).IsRequired();
         builder.Property(contract => contract.BindingFrom).IsRequired();
         builder.Property(contract => contract.ExpiringAt).IsRequired();
+
+        builder.OwnsMany<Annex>(nameof(BindingContract.RegisteredAnnexes), annex =>
```
Comment (kamilbaczek) [inline Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Infrastructure/Database/Configurations/BindingContractEntityConfiguration.cs:31]:
```csharp

public static class BindingContractExtensions
{
    public static void RegisterAnnexes(this EntityTypeBuilder<BindingContract> builder)
    {
        builder.OwnsMany<Annex>(nameof(BindingContract.RegisteredAnnexes), annex =>
        {
            annex.WithOwner().HasForeignKey(a => a.BindingContractId);
            annex.ToTable("Annexes");
            annex
                .Property(annex => annex.Id)
                .HasConversion(
                    id => id.Value,
                    value => new AnnexId(value));
            annex.Property(annex => annex.Id).IsRequired();
            annex.Property(annex => annex.BindingContractId)
                .HasConversion(
                    id => id.Value,
                    value => new BindingContractId(value));
            annex.Property(annex => annex.BindingContractId).IsRequired();
            annex.Property(annex => annex.ValidFrom).IsRequired();
        });
    }
}
```

```csharp
        builder.RegisterAnnexes()
```
what you think about extracting this code

**Review by kamilbaczek — CHANGES_REQUESTED** [review]

File: Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/BindingContract.cs (hunk 1 above)
Comment (meaboutsoftware) [reply Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/BindingContract.cs]: Great input, love it!

File: Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/BindingContract.cs (hunk 2 above)
Comment (meaboutsoftware) [reply Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/BindingContract.cs]: Done!

File: Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Infrastructure/Database/Configurations/BindingContractEntityConfiguration.cs (hunk 3 above)
Comment (meaboutsoftware) [reply Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Infrastructure/Database/Configurations/BindingContractEntityConfiguration.cs]: Done

**Review by kamilbaczek — APPROVED** [review]

**Pull request merged**